examples/mnist_denoising_autoencoder.py

```diff
@@ -0,0 +1,139 @@
+'''Trains a denoising autoenconder on MNIST dataset.
+
+Denoising is one of the classic applications of autoencoders.
+The denoising process removes unwanted noise that corrupted the 
+true signal. 
+
+Noise + Data ---> Denoising Autoencoder ---> Data
+
+Given a training dataset of corrupted data as input and 
+true signal as output, a denoising autoencoder can recover the 
+hidden structure to generate clean data.
+
+This example has modular design. The encoder, decoder and autoencoder
+are 3 models that share weights. For example, after training the 
+autoencoder, the encoder can be used to  generate latent vectors
+of input data for low-dim visualization like PCA or TSNE.
+'''
+
```

Comment:
Add imports to make the script compatible with py2/3:

```python
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
```

Reply:
Fixed. Thanks.